src/lib/nvme_pt_ioctl.c

```diff
@@ -2039,53 +2064,29 @@ int opal_eraselr_pt(struct sed_device *dev, const char *password,
 	return ret;
 }
 
-int opal_ds_admin_write(struct sed_device *dev, const char *key, uint8_t key_len, const void *from, uint32_t size, uint32_t offset)
+int opal_ds_read(struct sed_device *dev, enum SED_AUTHORITY auth,
+		const struct sed_key *key, uint8_t *to, uint32_t size,
+		uint32_t offset)
 {
-	int ret = 0;
-	struct sed_key disk_key;
+	int ret = 0, opal_auth;
 	struct opal_device *opal_dev;
 
-	if (key == NULL || from == NULL) {
-		SEDCLI_DEBUG_MSG("Must provide the password and a valid source pointer\n");
+	if (to == NULL) {
+		SEDCLI_DEBUG_MSG("Must provide a valid destination pointer\n");
 		return -EINVAL;
 	}
```

Comment:
One more check needs to be included here:
if (auth != SED_ANYBODY && key == NULL)
PRINT "User must provide a valid key"

Reply:
this check has been added to sed.c